src/java.base/share/classes/sun/security/ssl/SSLSessionContextImpl.java

```diff
@@ -87,14 +88,17 @@ final class SSLSessionContextImpl implements SSLSessionContext {
         cacheLimit = getDefaults(server);    // default cache size
 
         // use soft reference
-        sessionCache = Cache.newSoftMemoryCache(cacheLimit, timeout);
-        sessionHostPortCache = Cache.newSoftMemoryCache(cacheLimit, timeout);
         if (server) {
+            sessionCache = Cache.newSoftMemoryCache(cacheLimit, timeout);
+            sessionHostPortCache = Cache.newSoftMemoryCache(cacheLimit, timeout);
```

Comment:
preexisting, but I wonder if we could use nullCache here and in client's sessionCache

Reply:
Let's deal with that outside of this RFE, but there are tests that use it and I'm pretty sure this has public API usage with SSLSessionContext.getSession(byte[])